packages/persist-state/src/lib/persist-state.ts

```diff
@@ -40,13 +48,19 @@ export function persistState<S extends Store>(store: S, options: Options<S>) {
   const initialized = new ReplaySubject<boolean>(1);
 
   const loadFromStorageSubscription = from(
-    storage.getItem(merged.key!)
+    storage.getItem(merged.key!),
   ).subscribe((value) => {
     if (value) {
+      const preparedValue = merged.preStoreValueInit
+        ? merged.preStoreValueInit(value)
+        : isRecord(value)
+          ? value
+          : { value };
```

Comment:
why it's { value }?

Reply:
In case of storing compressed / encoded string directly in local storage.

```
preStoreValueInit?: (value: Record<string, any> | string) => Partial<StoreValue<S>>;
preStorageValueUpdate?: (storeName: string, state: Partial<StoreValue<S>>) => Partial<StoreValue<S>> | string;
```

`Record<string, any> | string`

Reply:
Later on, to support spread operator, the potential string value gets wrapped in the object.

Reply:
but that's not the store value, no?

Reply:
```js
      store.update((state) => {
        return merged.preStoreInit!({
          ...state,
          ...preparedValue,
        });
      });
```

Reply:
Essentially, this is added to support both `Record<string, any>` and `string` types in synchronisation between store and storage.

Reply:
It is a value parsed from local storage, prior to initialising the store.